app/assets/javascripts/discourse/app/templates/components/user-nav.hbs

```diff
@@ -0,0 +1,252 @@
+<section class="user-primary-navigation">
+  <ul class="main-nav nav nav-pills user-nav" data-count={{this.navCount}}>
+    {{#unless @model.profile_hidden}}
+      {{#if this.site.mobileView}}
+        {{#if this.currentUser.staff}}
+          <li class="user-primary-navigation_item"><a href={{@model.adminPath}}>{{d-icon "wrench"}}<span class="d-button-label">{{i18n "admin.user.show_admin_profile"}}</span></a></li>
+        {{/if}}
+      {{/if}}
+
+      <li class="summary user-primary-navigation_item">
+        <LinkTo @route="user.summary">
+          {{d-icon "user"}}
+          <span>{{i18n "user.summary.title"}}</span>
+        </LinkTo>
+      </li>
+
+      <li role="button" class="activity user-primary-navigation_item-parent" {{on "click" this.toggleSubmenu}}>
+        <a class={{if (eq @currentRouteParent "userActivity") "active" }}>
+          {{d-icon "stream"}}<span>{{i18n "user.activity_stream"}}</span>{{d-icon "chevron-down"}}
+        </a>
+
+        <ul class="user-primary-navigation_submenu activity-nav">
+          <DNavigationItem @route="userActivity.index">{{i18n "user.filters.all"}}</DNavigationItem>
+          <DNavigationItem @route="userActivity.topics">{{i18n "user_action_groups.4"}}</DNavigationItem>
+          <DNavigationItem @route="userActivity.replies">{{i18n "user_action_groups.5"}}</DNavigationItem>
+
+          {{#if @user.showRead}}
+            <DNavigationItem @route="userActivity.read" @title={{i18n "user.read_help"}}>
+              {{i18n "user.read"}}
+            </DNavigationItem>
+          {{/if}}
+
+          {{#if @user.showDrafts}}
+            <DNavigationItem @route="userActivity.drafts">
+              {{@draftLabel}}
+            </DNavigationItem>
+          {{/if}}
+
+          {{#if (gt @model.pending_posts_count 0)}}
+            <DNavigationItem @route="userActivity.pending">
+              {{@pendingLabel}}
+            </DNavigationItem>
+          {{/if}}
+
+          <DNavigationItem @route="userActivity.likesGiven">{{i18n "user_action_groups.1"}}</DNavigationItem>
+          {{#if @user.showBookmarks}}
+            <DNavigationItem @route="userActivity.bookmarks">{{i18n "user_action_groups.3"}}</DNavigationItem>
+          {{/if}}
+
+          <PluginOutlet @name="user-activity-bottom" @tagName="" @connectorTagName="" @args={{hash model=@model}} />
+        </ul>
+      </li>
+
+      <UserNav::DropdownList @class="activity" @isActive={{eq @currentRouteParent "userActivity"}} @icon="stream" @text={{i18n "user.activity_stream"}}>
```

Comment:
@awesomerobot I pushed a new component for the dropdown which relies on Ember to display/hide the dropdown instead of using CSS to display/hide the dropdown. Do feel free to let me know if anything is unclear about the component or if you need any help extending the component.